mindmeld/components/entity_resolver.py

```diff
@@ -569,31 +738,242 @@ def _predict_exact_match(self, entity):
+        similarity_scores = np.around(similarity_scores, decimals=2)
+
+        # results in descending scores
+        return sorted(list(zip(synonyms, similarity_scores)), key=lambda x: x[1], reverse=True)
+
+    @staticmethod
+    def _process_entity_map(entity_type, entity_map):
+        """Loads in the mapping.json file and stores the synonym mappings in a item_map and a
+        synonym_map for exact match entity resolution when Elasticsearch is unavailable
+
+        Args:
+            entity_type: The entity type associated with this entity resolver
+            entity_map: The loaded mapping.json file for the given entity type
+        """
+        item_map = {}
+        syn_map = {}
+        seen_ids = []
+        for item in entity_map.get("entities"):
+            cname = item["cname"]
+            item_id = item.get("id")
+            if cname in item_map:
+                msg = "Canonical name %s specified in %s entity map multiple times"
+                logger.debug(msg, cname, entity_type)
+            if item_id:
+                if item_id in seen_ids:
+                    msg = "Item id {!r} specified in {!r} entity map multiple times"
+                    raise ValueError(msg.format(item_id, entity_type))
+                seen_ids.append(item_id)
+
+            aliases = [cname] + item.pop("whitelist", [])
+            items_for_cname = item_map.get(cname, [])
+            items_for_cname.append(item)
+            item_map[cname] = items_for_cname
+            for norm_alias in aliases:
+                if norm_alias in syn_map:
+                    msg = "Synonym %s specified in %s entity map multiple times"
+                    logger.debug(msg, cname, entity_type)
+                cnames_for_syn = syn_map.get(norm_alias, [])
+                cnames_for_syn.append(cname)
+                syn_map[norm_alias] = list(set(cnames_for_syn))
+
+        return {"items": item_map, "synonyms": syn_map}
+
+    def _fit(self, clean):
+        """
+        Fits the resolver model
+
+        Args:
+            clean (bool): If ``True``, deletes existing dump of synonym embeddings file
+        """
+
+        # load model
         try:
-            if self._use_text_rel:
-                scoped_index_name = get_scoped_index_name(
-                    self._app_namespace, self._es_index_name
-                )
-                if not self._es_client.indices.exists(index=scoped_index_name):
-                    self.fit()
-            else:
-                self.fit()
+            _encoding_type = (
+                self.er_config.get("model_settings", {})
+                    .get("encoding_type", "mean")
+            )
+            transformer_model = Transformer(self._sbert_model_pretrained_name_or_abspath)
+            pooling_model = Pooling(transformer_model.get_word_embedding_dimension(),
+                                    pooling_mode_cls_token=_encoding_type == "cls",
```

Comment:
Is it correct to say a boolean has to be passed to `pooling_mode_cls_token `?

Reply:
I think yes.